### PR TITLE
[12.0][ADD] hr_employee_department_secondary

### PR DESCRIPTION
**Pull request by oriolpiera**
Add secondary department feature on employee page. In this new field `secondary_department_ids` you can add secondary departments where employee works, excpet de main department defined in `department_id`.

Also we added a new filter in "All Leaves" named "All My Secondary Department's Leaves"  where you can see all the holidays for all the co-workers in your secondary departments.

image
image

### CODE REVIEW

**Comment by Saran440** [comment]
@oriolpiera 
Hi, I think this module should be module name 'hr_employee_department_secondary'
and concept similar with module `partner_industry_secondary`. it will helpful for adapte to your module.

https://github.com/OCA/partner-contact/tree/12.0/partner_industry_secondary

**Comment by oriolpiera** [comment]
@Saran440 sorry for the delay. We just apply your suggestions.

**Comment by nimarosa** [comment]
@oriolpiera Please rebase the PR.

**Comment by oriolpiera** [comment]
@nimarosa done

File: hr_employee_department_secondary/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+    'name': "HR Employee Departments secondary",
+    'version': '12.0.0.0.4',
+    'depends': [
+        'hr',
+        'hr_holidays',
```
Comment (hieulucky111) [inline hr_employee_department_secondary/__manifest__.py:6]: i don't think we should depends on this module.

File: hr_employee_department_secondary/models/hr_employee.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+from odoo import api, exceptions, fields, models, _
+
+
+class Employee(models.Model):
+    _inherit = ['hr.employee']
```
Comment (hieulucky111) [inline hr_employee_department_secondary/models/hr_employee.py:5]: should we use `[]`?

File: hr_employee_department_secondary/models/hr_employee.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    @api.constrains('department_id', 'secondary_department_ids')
+    def _check_departments(self):
+        for employee in self:
+            if employee.department_id in employee.secondary_department_ids:
+                raise exceptions.ValidationError(
+                    _('The main department must be different '
+                      'from the secondary departments.'))
+
+class Department(models.Model):
+    _inherit = ['hr.department']
```
Comment (hieulucky111) [inline hr_employee_department_secondary/models/hr_employee.py:22]: same above.

File: hr_employee_department_secondary/views/hr_leave_views.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record model="ir.ui.view" id="hr_view_leave_form_multidepartment">
+        <field name="name">hr.leave.form.inherit</field>
+        <field name="model">hr.leave</field>
```
Comment (hieulucky111) [inline hr_employee_department_secondary/views/hr_leave_views.xml:5]: i think you can create another module in `hr-holidays` repository and depends on this module.

**Review by hieulucky111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**